ping.go

```diff
@@ -339,6 +339,58 @@ func (p *Pinger) Addr() string {
 	return p.addr
 }
 
+// SetSrcAddr set source by interface name or ipv4 address
+// Interface name can be "eth0" or ip like "192.168.1.239"
+func (p *Pinger) SetSrcAddr(iface string) error {
+	if ip := net.ParseIP(iface); ip != nil {
+		if ip.To4() != nil {
+			p.SetNetwork("ipv4")
+		} else {
+			p.SetNetwork("ipv6")
+		}
+		p.Source = iface
+		return nil
+	}
+	if ipAddr, err := p.GetIfaceAddrs(iface); err != nil {
+		return err
+	} else {
+		p.Source = ipAddr[0].String() // As the private IP range is generally in ipv4, so as default it sets ipv4 of the iface
```

Comment:
`ipAddr` is an array but we always take the first value - this feels like a likely source of subtle bugs for end users. Not saying it isn't the least-worst thing to do, but maybe we should document. As an aside, the comment is a bit misleading here - there's no guarantee this IP will be a private one, especially if it's a v6 address.

Reply:
Cool, I will make the changes and commit it. Thanks for the review 🎉